### PR TITLE
fix: delete certificate properties of payload for correction

### PR DESCRIPTION
**Pull request by Nil20**

### CODE REVIEW

File: packages/client/src/declarations/submissionMiddleware.ts [hunk 1]
```diff
@@ -187,6 +187,7 @@ export const submissionMiddleware: Middleware<{}, IStoreState> =
       graphqlPayload.registration.correction =
         declaration.data.registration.correction ?? {}
       graphqlPayload.registration.correction.values = changedValues
+      delete graphqlPayload.registration.certificates
```
Comment (Zangetsu101) [inline packages/client/src/declarations/submissionMiddleware.ts:190]: @Nil20 Could you please add a comment explaining why we are doing this?

Comment (naftis) [reply packages/client/src/declarations/submissionMiddleware.ts:191]: so if we're submitting a correction, we remove certificate properties here, what is the reason for it though?

Comment (Nil20) [reply packages/client/src/declarations/submissionMiddleware.ts:191]: The correction request payload doesn't have any `certificates` properties in it.
https://github.com/opencrvs/opencrvs-core/blob/c19bf79ee14d178ee6e22fb596ecd76ef21ca369/packages/workflow/src/records/validations.ts#L76

**Comment by Zangetsu101** [comment]
@Nil20 Perhaps it makes more sense to remove the certificate values from here: https://github.com/opencrvs/opencrvs-core/blob/develop/packages/client/src/views/PrintCertificate/usePrintableCertificate.ts#L200 before sending to the correction flow?

File: packages/client/src/views/PrintCertificate/usePrintableCertificate.ts [hunk 2]
```diff
@@ -196,10 +196,20 @@ export const usePrintableCertificate = (declarationId: string) => {
...
 
-  const handleEdit = () =>
+  const handleEdit = () => {
+    // Delete certificate properties during print record corrections
+    // since correction flow doesn't handle certificates
+    if (declaration?.data?.registration.certificates) {
+      delete (declaration?.data?.registration as { certificates: any })
+        ?.certificates
+      dispatch(modifyDeclaration(declaration))
+      dispatch(writeDeclaration(declaration))
```
Comment (Zangetsu101) [inline packages/client/src/views/PrintCertificate/usePrintableCertificate.ts:206]: We should maintain immutability when updating the declaration. Although it probably won't affect anything now, might save us from tearing our hair out due to some bug in the future. :wink:

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**